linetools/spectra/xspectrum1d.py

```diff
@@ -772,12 +781,106 @@ def fit_continuum(self, knots=None, edges=None, wlim=None, dw=10.,
             [tuple(pts) for pts in wrapper.contpoints])
         self.meta['contpoints'].sort()
 
+    def _interp_continuum(self, x, y):
+        """
+        Returns an interpolation of the continuum using knots at
+        (x,y) positions. If at least 5 points exists, it uses Akima
+        Spline interpolation. Else, it uses linear interpolation.
+
+        Parameters
+        ----------
+        x : array, shape(N,)
+            The x positions of the knots to interpolate through
+            Assumed to be in Angstroms.
+        y : array, shape(N,)
+            The y positions of the knots to interpolate through
+
+        Returns
+        -------
+        co : array, shape(M,)
+            Values of the interpolated continuum for each self.wavelength point
+
+        """
+
+        if len(y) >= 5:
+            # need 5 points to define an Akima Spline
+            spline = AkimaSpline(x, y)
+            co = spline(self.wavelength.value)
+        else:
+            co = np.interp(self.wavelength.value, x, y)
+
+        return co
+
+    def perturb_continuum(self, rel_var=0.05, seed=None):
+        """
+        Adds a perturbation to the continuum by adding noise to the contpoints.
+        This may be useful for estimating the uncertainty in the continuum.
+
+        Parameters
+        ----------
+        rel_var : float, optional
+            Relative variation in flux w/r to its original value. This will be
+            the standard deviation of a Normal distribution added to
+            the contpoints' `y` values. Default is rel_var=0.05 (i.e. 5%).
+        seed : int, optional
+            The seed of the (pseudo)random number generator.
+
+        Returns
+        -------
+        Updates self.co with a perturbation.
+        Note: To reset to the original continuum use reset_continuum()
+
+        """
+        try:
+            contpoints = self.meta['contpoints']
+        except:
+            raise RuntimeError('Meta data does not exist; cannot perturb the continuum.')
+        xy = np.array(contpoints)
+        xy = xy.transpose()
+        x, y = xy[0], xy[1]
+
+        # Get indices in the full spectrum
+        # inds = np.array([np.where(np.fabs(x[i]-self.dispersion.value)<0.01)[0][0] for i in range(len(x))])
+        # y_err = y_err = self.uncertainty.array[inds]
+
+        # Seed
+        np.random.seed(seed=seed)
+        #add Normal noise to y points
+        y += np.random.normal(0, y * rel_var, len(y))
```

Comment:
I think this line doesn't modify the original meta['contpoints'] because it's copied by the `xy = np.array(contpoints)` above. But can you change this `y = y + np.random.normal(...` just make sure?